extensions/kubernetes-client/runtime/src/main/java/io/quarkus/kubernetes/client/runtime/graal/KubernetesDeserializerSubstitutions.java

```diff
@@ -0,0 +1,17 @@
+package io.quarkus.kubernetes.client.runtime.graal;
+
+import com.oracle.svm.core.annotate.Substitute;
+import com.oracle.svm.core.annotate.TargetClass;
+
+@TargetClass(className = "io.fabric8.kubernetes.internal.KubernetesDeserializer")
+public final class KubernetesDeserializerSubstitutions {
+
+    @Substitute
+    private static Class loadClassIfExists(String className) {
```

Comment:
Please add a proper comment explaining the why of this substitution. I suppose it's related to the class loader, somehow?

Furthermore, it doesn't look related to the first part of the commit? Should it be a separate commit explaining what it's supposed to fix?

Reply:
Sure thing, I will add a comment. But I think that it should be one commit, since this change is also needed to get the `Watcher` working.

Reply:
Furthermore this part of the code seems to only be reached on watch operations, not crud operations

Reply:
Added comment